### ISSUE TITLE
Width of U+2009 THIN SPACE may need to be modified for French

### ISSUE THREAD

**Issue by roozbehp** [google/roboto#24]
According to Microsoft's [Character design standards](https://www.microsoft.com/typography/developers/fdsspec/spaces.aspx), the width of U+2009 THIN SPACE is language-dependent. Here's the actual text:

> *Thin space* U+2009 - Standard setting is 1/5 of the em space. 410 units in a 2048 unit per em font. This should be language dependent. The standard language dependent setting for French is 1/8 of the em space. 256 units in a 2048 unit per em font.
>
> *Note* : When traditionally typesetting he French language a word space is inserted before or after several punctuation characters. These characters are *colon*, *semi colon*, *question*, *exclamation*, *right guillemets*, and *left guillemets*. Commonly the preferred word space used is a thin space of 1/8 the em. Some French typographers prefer to use a larger space character of 1/4 the em with the colon and some other punctuation characters. OpenType supports character substitution and language dependant variants.

/cc @christianrobertson, @raphlinus

**Comment by xiangyexiao** [comment]
assign it to myself to contact the contractor and see if they can fix this.

**Comment by jamesgk** [comment]
Fixed by latest update, #97

**Comment by adrientetar** [comment]
Are there binaries available for the latest delivery?

**Comment by jamesgk** [comment]
Yea, the TTFs in `/hinted` have been updated.

**Comment by adrientetar** [comment]
Thanks – change looks good.

**Comment by jamesgk** [comment]
Great! Thanks Adrien.

**Comment by moyogo** [comment]
Sorry, I meant to mention this earlier, but the correct thin space used in French is supposed to be a non breaking thin space: U+202F narrow no-break space. Pretty much all the French typography guides say the thin space is non breaking.
Thin space U+2009 breaks and a line starting with a punctuation sign is not good.
So I suppose both U+202F and U+2009 should have the same advance width.

I’m not sure either where this “standard” of 1/8 em comes from, but it probably doesn’t matter at this point.

*Lexique des règles typographiques en usage à l’Imprimerie nationale* (1990), *Code typographique* (1993), *Autour des mots* (2005), *Le Ramat européen de la typographie* (2009) do not mention how wide the thin space is supposed to be but recommend its use in front of double punctuation (except colon), *Le style du Monde* recommends it for all double punctuation without mentioning its width either.

*Le Ramat de la typographie* (2008) mentions 40% of the space when trying to simulate the thin space.
*Dictionnaire des règles typographique* (2010) mentions that the 1/4 em is also called thin space
*Dictionnaire typographique* (2005) gives 1.5 pt for the thin space, 2 pt or 2.5 pt for the half space and 4 pt for the full space and then recommends actually using the half space (where one uses the thin space in the other guides).
*Orthotypo* (2005) mentions thin space 1 pt, pre-thin space 1.5 pt, half space 1/4 em and space 1/3 em.
*Mémento typographique* (1998) mentions “at least 2 pt” but says one can settle for 1/3 em when needed.
*Orthotypo & Co* (2013) also says the thin space is 1/4 em.

So basically, depending on where one looks, a French thin space can be 1/8, 1/6, 1/4, or even 1/3 em. The normal space’s width doesn’t seem to be standard either.

**Comment by adrientetar** [comment]
> I’m not sure either where this “standard” of 1/8 em comes from, but it probably doesn’t matter at this point.

I find it boils down to personal preference amongst French typographers, though French-made fonts/books generally have tighter than other countries – the French-made fonts I have have smaller thinsp, few have U+202F though it’s highly useful! (I suspect InDesign makes non-breaking space regardless of whether you use the non-breaking codepoint, otherwise more French fonts would have it).
